### ISSUE TITLE
Exception occurs when performing type completion for some variables.

### ISSUE THREAD

**Issue by mir-am** [saltudelft/type4py-vscode-ext#7]
For the file `workers.py` from [here](https://github.com/ml-libs/mlserve/blob/master/mlserve/worker.py#L46), when performing type completion at line 46 for the variable `results_list`, the following exception occurs:
```
Exception has occurred: TypeError: Cannot read property 'forEach' of undefined
t ParamHintCompletionProvider.<anonymous> (/Users/amir/projects/type4py-vscode-ext/dist/extension.js:250:99)
	at Generator.next (<anonymous>)
	at /Users/amir/projects/type4py-vscode-ext/dist/extension.js:212:71
	at new Promise (<anonymous>)
	at /Users/amir/projects/type4py-vscode-ext/dist/extension.js:208:12
	at ParamHintCompletionProvider.provideCompletionItems (/Users/amir/projects/type4py-vscode-ext/dist/extension.js:235:16)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:171333
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:64:11884
	at new Promise (<anonymous>)
	at Object.g [as asPromise] (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:64:11856)
	at F.provideCompletionItems (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:171304)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:190970
	at ze._withAdapter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:181033)
	at ze.$provideCompletionItems (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:84:190946)
	at d._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:88:12815)
	at d._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:88:12507)
	at d._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:88:11108)
	at d._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:88:9960)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:88:8064
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:57:1835)
	at E.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:65:15634)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:103:29816
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:57:1835)
	at E.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:65:15634)
	at t._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:65:20892)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:65:17778
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:57:1835)
	at c.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:65:12995)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:65:12343
	at Socket.b (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:103:13287)
	at Socket.emit (events.js:315:20)
	at addChunk (internal/streams/readable.js:309:12)
	at readableAddChunk (internal/streams/readable.js:284:9)
	at Socket.push (internal/streams/readable.js:223:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
	at Pipe.callbackTrampoline (internal/async_hooks.js:131:14)
```

**Comment by elatoskinas** [comment]
This seems to be a result of the variable line numbers only representing the last instance when they occur more than once in a function (because a map is used). Minimum reproducible example:

```
def fun(a):
    if a:
        x = 1
    else:
        x = 3
    
    return x
```

The annotation will work on `x = 3`, but not on `x = 1`, because the map only tracks the last `x = 3`.

Good catch! We should either fix this by capturing all occurrences via line numbers on the server and updating the client, or delegating this to the fully-qualified name/cache task.

**Comment by elatoskinas** [comment]
I have added the sample file to the repo: https://github.com/saltudelft/type4py-vscode-ext/commit/28d88e33f9f4544fd5b752bc10416ad5a3e97873

**Comment by mir-am** [comment]
Addressed in #18.
